tests/system/data_sources/common_functions.py

```diff
@@ -346,17 +342,56 @@ def generate_partitions_test(
     config_managers = main.build_config_managers_from_args(args, consts.ROW_VALIDATION)
     partition_builder = PartitionBuilder(config_managers, args)
     partition_filters = partition_builder._get_partition_key_filters()
-    yaml_configs_list = partition_builder._add_partition_filters(partition_filters)
 
     assert len(partition_filters) == 1  # only one pair of tables
     # Number of partitions is as requested - assume table rows > partitions requested
     assert len(partition_filters[0][0]) == partition_builder.args.partition_num
     assert partition_filters[0] == expected_filter
 
-    # Next, that the partitions were split into the files correctly
-    # 2 files were created with upto 5 validations in each file
-    assert len(yaml_configs_list[0]["yaml_files"]) == 2
-    # 5 validations in the first file
-    assert len(yaml_configs_list[0]["yaml_files"][0]["yaml_config"]["validations"]) == 5
-    # 4 validations in the second file
-    assert len(yaml_configs_list[0]["yaml_files"][1]["yaml_config"]["validations"]) == 4
+
+def partition_query_test(
+    expected_filter: str,
+    tmp_path,
+    pk="course_id,quarter_id,student_id",
+    tables="pso_data_validator.test_generate_partitions",
+    filters="quarter_id != 1111",
+):
+    """Test generate table partitions for custom queries. Usually only the partition_filter is different
+    because of the differences in SQL between the databases. Some databases have different table names,
+    Teradata has a different syntax for inequality and Postgres has different column names/types for primary keys.
+    The unit tests in tests/unit/test_partition_builder.py check if the filters are split into configs and
+    stored in the filesystem correctly.
+    """
+    tables_list = tables.split("=")
+    source_table_name = tables_list[0]
+    target_table_name = tables_list[1] if len(tables_list) == 2 else tables_list[0]
+    target_query = f"select * from {target_table_name}"
+    source_query = f"select * from {source_table_name}"
+    source_query_file = tmp_path / "source_query.sql"
```

Comment:
I asked about the data type above because it's not clear to me what's happening at this line. 

I tested on my own and I got:
```
>>> tmp_path = "/teste"
>>> print(tmp_path / "source_query.sql")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for /: 'str' and 'str'
```